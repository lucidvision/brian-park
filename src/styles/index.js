export const colors = {
  primary: 'rgba(0, 13, 85, 0.8)',
  black: '#000',
  white: '#fff',
  gradientLight: 'rgba(0, 38, 255, 0.8)',
  gradientDark: 'rgba(0, 0, 0, 0.8)'
}

export const fontSizes = {
  large: '6rem',
  medium: '4rem',
  small: '2.2rem',
  default: '1.6rem'
}

export const screenSizes = {
  desktop: '112.5em',
  tablet: '75em',
  phone: '56.25em'
}

export const utilities = {
  marginSmall: '1.5rem',
  marginMedium: '4rem',
  marginLarge: '8rem'
}
