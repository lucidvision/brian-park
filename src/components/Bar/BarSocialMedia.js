import React from 'react'
import styled from 'styled-components'
import { IconLink } from 'components'
import { colors } from 'styles'
import _ from 'lodash'

const StyledDiv = styled.div`
  width: 40rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

const links = [
  {
    className: 'fab fa-linkedin',
    url: 'https://www.linkedin.com/in/lucidvision/'
  },
  {
    className: 'fab fa-angellist',
    url: 'https://angel.co/brian-park-11'
  },
  {
    className: 'fab fa-github',
    url: 'https://github.com/lucidvision'
  },
  {
    className: 'fab fa-facebook',
    url: 'https://www.facebook.com/Lucid.Vision'
  },
  {
    className: 'fa fa-envelope',
    url: 'mailto:brianpark25@gmail.com'
  }
]

export default function BarSocialMedia (props) {
  return (
    <StyledDiv>
      {_.map(links, (link, index) => <IconLink color={colors.white} key={index} {...link} />)}
    </StyledDiv>
  )
}
