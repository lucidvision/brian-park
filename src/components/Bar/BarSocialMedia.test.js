import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import BarSocialMedia from './BarSocialMedia'
import { colors } from 'styles'

Enzyme.configure({adapter: new Adapter()})

const links = [
  {
    className: 'fab fa-linkedin',
    color: colors.white,
    url: 'https://www.linkedin.com/in/lucidvision/'
  },
  {
    className: 'fab fa-angellist',
    color: colors.white,
    url: 'https://angel.co/brian-park-11'
  },
  {
    className: 'fab fa-github',
    color: colors.white,
    url: 'https://github.com/lucidvision'
  },
  {
    className: 'fab fa-facebook',
    color: colors.white,
    url: 'https://www.facebook.com/Lucid.Vision'
  },
  {
    className: 'fa fa-envelope',
    color: colors.white,
    url: 'mailto:brianpark25@gmail.com'
  }
]

describe('BarSocialMedia', () => {
  const barSocialMedia = shallow(<BarSocialMedia />)

  it('renders correctly', () => {
    expect(barSocialMedia).toMatchSnapshot()
  })

  describe('Social Media Links', () => {
    it('renders 5 links', () => {
      expect(barSocialMedia.find('IconLink').length).toEqual(5)
    })

    links.forEach((props, index) => {
      it(`renders the ${props.url} link`, () => {
        expect(barSocialMedia.find('IconLink').at(index).props()).toEqual(props)
      })
    })
  })
})
