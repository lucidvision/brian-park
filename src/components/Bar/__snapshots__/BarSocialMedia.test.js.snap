// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarSocialMedia renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BarSocialMedia />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <IconLink
          className="fab fa-linkedin"
          color="#fff"
          url="https://www.linkedin.com/in/lucidvision/"
/>,
        <IconLink
          className="fab fa-angellist"
          color="#fff"
          url="https://angel.co/brian-park-11"
/>,
        <IconLink
          className="fab fa-github"
          color="#fff"
          url="https://github.com/lucidvision"
/>,
        <IconLink
          className="fab fa-facebook"
          color="#fff"
          url="https://www.facebook.com/Lucid.Vision"
/>,
        <IconLink
          className="fa fa-envelope"
          color="#fff"
          url="mailto:brianpark25@gmail.com"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "className": "fab fa-linkedin",
          "color": "#fff",
          "url": "https://www.linkedin.com/in/lucidvision/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "className": "fab fa-angellist",
          "color": "#fff",
          "url": "https://angel.co/brian-park-11",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "className": "fab fa-github",
          "color": "#fff",
          "url": "https://github.com/lucidvision",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "className": "fab fa-facebook",
          "color": "#fff",
          "url": "https://www.facebook.com/Lucid.Vision",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "className": "fa fa-envelope",
          "color": "#fff",
          "url": "mailto:brianpark25@gmail.com",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <IconLink
            className="fab fa-linkedin"
            color="#fff"
            url="https://www.linkedin.com/in/lucidvision/"
/>,
          <IconLink
            className="fab fa-angellist"
            color="#fff"
            url="https://angel.co/brian-park-11"
/>,
          <IconLink
            className="fab fa-github"
            color="#fff"
            url="https://github.com/lucidvision"
/>,
          <IconLink
            className="fab fa-facebook"
            color="#fff"
            url="https://www.facebook.com/Lucid.Vision"
/>,
          <IconLink
            className="fa fa-envelope"
            color="#fff"
            url="mailto:brianpark25@gmail.com"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "className": "fab fa-linkedin",
            "color": "#fff",
            "url": "https://www.linkedin.com/in/lucidvision/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "className": "fab fa-angellist",
            "color": "#fff",
            "url": "https://angel.co/brian-park-11",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "className": "fab fa-github",
            "color": "#fff",
            "url": "https://github.com/lucidvision",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "className": "fab fa-facebook",
            "color": "#fff",
            "url": "https://www.facebook.com/Lucid.Vision",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "className": "fa fa-envelope",
            "color": "#fff",
            "url": "mailto:brianpark25@gmail.com",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
